File: python/2609.py
# 내 풀이
a, b = map(int, input().split())


def gcd(a, b):
    k = 2
    result = 1
    c = min(a, b)

    while(k <= c):
        if(a % k == 0 and b % k == 0):
            a = a//k
            b = b//k
            result *= k
        else:
            k += 1
    return result


def lcm(a, b):
    g = gcd(a, b)
    return(a*b//g)


print(gcd(a, b))
print(lcm(a, b))

# 유클리드 호제법 풀이

# a, b = map(int, input().split())


# def gcd(a, b):
#     if b == 0:
#         return a
#     else:
#         return gcd(b, a % b)


# def lcm(a, b):
#     g = gcd(a, b)
#     return int(a*b//g)


# print(gcd(a, b))
# print(lcm(a, b))

File: c++/2609.cpp
// #include <iostream>

// using namespace std;

// int main(){
//   int a, b, d = 2;
//   int lcm, gcd = 1;
//   cin >> a >> b;

//   while(!(d > a || d > b)){
//     if(a%d==0 && b%d==0){
//       a /= d;
//       b /= d;
//       gcd *= d;
//     } else{
//       d += 1;
//     }
//   }
//   lcm = gcd*a*b;

//   cout << gcd << '\n';
//   cout << lcm << '\n';

//   return 0;
// }

#include <iostream>

using namespace std;

int gcd(int a, int b){
  int c;
  while(b!=0){
    c = a%b;
    a = b;
    b = c;
  }
  return a;
}

int main(){
  int m, n;
  cin >> m >> n;

  int g = gcd(m,n);
  int lcm = m*n/g;

  cout << g << '\n';
  cout << lcm << '\n';

  return 0;
}